packages/react/src/useComponentValue.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, ComponentValue, EntityIndex, getComponentValue, Has, Metadata, Schema } from "@latticexyz/recs";
+import { useEntityQuery } from "./useEntityQuery";
+
+export function useComponentValue<S extends Schema>(
+  entityIndex: EntityIndex | undefined,
+  component: Component<S, Metadata, undefined>,
+  defaultValue: ComponentValue<S>
+): ComponentValue<S>;
+
+export function useComponentValue<S extends Schema>(
+  entityIndex: EntityIndex | undefined,
+  component: Component<S, Metadata, undefined>
+): ComponentValue<S> | undefined;
+
+export function useComponentValue<S extends Schema>(
+  entityIndex: EntityIndex | undefined,
+  component: Component<S, Metadata, undefined>,
+  defaultValue?: ComponentValue<S>
+) {
+  const entities = useEntityQuery([Has(component)]);
```

Comment:
would this hook cause a react rerender every time any value in the component changes or only if the value of the entity specified in the `entityIndex` param changes?

Reply:
Ohh this is a great point. It'd currently reevaluate for anything matching the query. I may rewrite this to not depend on `useEntityQuery` and be more mindful of rendering only when the single entity value changes.